src/box/lua/console.lua

```diff
@@ -436,6 +524,18 @@ local function local_eval(storage, line)
     if not res[1] and not in_txn_before and ffi.C.box_txn() then
         box.rollback()
     end
+    -- Check trace is correct. Skip check for non ClienError.
```

Comment:
Fixed. 

```diff
--- a/src/box/lua/console.lua
+++ b/src/box/lua/console.lua
@@ -524,10 +524,10 @@ local function local_eval(storage, line)
     if not res[1] and not in_txn_before and ffi.C.box_txn() then
         box.rollback()
     end
-    -- Check trace is correct. Skip check for non ClienError.
+    -- Check trace is correct.
     if not res[1] and check_trace then
         local err = res[2]
-        if err.type == nil or err.trace == nil then
+        if type(err) ~= 'cdata' or err.trace == nil then
             return format(false, string.format('Non box.error: %s', err))
         end
         local trace = err.trace[1]
```

Reply:
It may not work as expected if someone throws a cdata object that doesn't have index, e.g. `1LL`. Let's add `box.error.is` that would check that the FFI type, similarly to `box.tuple.is` (in a separate commit).

Reply:
Fixed.

```diff
diff --git a/src/box/lua/console.lua b/src/box/lua/console.lua
index 42b0ff72b..df5ce9208 100644
--- a/src/box/lua/console.lua
+++ b/src/box/lua/console.lua
@@ -504,7 +504,7 @@ local function local_eval(storage, line)
     if not res[1] and
        tarantool._internal.trace_check_is_required(res[2].info.short_src) then
         local err = res[2].err
-        if type(err) ~= 'cdata' or err.trace == nil then
+        if not box.error.is(err) then
             return format(false, string.format('Non box.error: %s', err))
         end
         local trace = err.trace[1]
```